### PR TITLE
Feature: Add `Integer` type and arithmetic methods.

### PR DESCRIPTION
**Pull request by MajorLift**
Implements `Integer` kind category and associated arithmetic methods, as well as new methods for other numeric categories.

## Notes

- Distinguished `Integer.Modulo` and `Integer.Remainder`
  - `Remainder` implements the `%` operator in JavaScript.
    - Same behavior as `Modulo` methods in `NaturalNumber`, `DigitList`.
  - `Modulo` is `((n % d) + d) % d`.

```ts
$<$<Integer.Remainder, 10>, 3>; // 1
$<$<Integer.Remainder, 10>, -3>; // -1
$<$<Integer.Remainder, -10>, 3>; // -1
$<$<Integer.Remainder, -10>, -3>; // 1

$<$<Integer.Modulo, 10>, 3>; // 1
$<$<Integer.Modulo, 10>, -3>; // -2
$<$<Integer.Modulo, -10>, 3>; // 2
$<$<Integer.Modulo, -10>, -3>; // -1
```

- Fixed `Number.Absolute` not handling `bigint` correctly (previously returned 0 for negative inputs).

# List of Contents

### Arithmetic methods
- [x] IsLessThan
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### JSDocs
- [x] IsLessThan
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### Arithmetic methods
- [x] Increment
- [x] Decrement
- [x] Add
- [x] Subtract
- [x] SubtractBy
- [x] Multiply
- [x] Divide
- [x] DivideBy
- [x] Remainder
- [x] RemainderBy
- [x] Modulo
- [x] ModuloBy
- [x] IsEven
- [x] IsOdd
- [x] Compare
- [x] IsLessThan
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### JSDocs
- [x] Increment
- [x] Decrement
- [x] Add
- [x] Subtract
- [x] SubtractBy
- [x] Multiply
- [x] Divide
- [x] DivideBy
- [x] Remainder
- [x] RemainderBy
- [x] Modulo
- [x] ModuloBy
- [x] IsEven
- [x] IsOdd
- [x] Compare
- [x] IsLessThan
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### Methods
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### JSDocs
- [x] Increment
- [x] Decrement
- [x] Add
- [x] Subtract
- [x] SubtractBy
- [x] Divide
- [x] DivideBy
- [x] Multiply
- [x] IsEven
- [x] IsOdd
- [x] IsLessThanOrEqual
- [x] IsGreaterThan
- [x] IsGreaterThanOrEqual

### Methods
- [x] TrimRight

### JSDocs
- [x] TrimRight 
- [x] Increment
- [x] Multiply

### CODE REVIEW

File: src/integer/divide.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    A_LIST,
+    B_LIST
+  >,
+  QUOTIENT extends Number.Number = DigitList._$toNumber<QUOTIENT_LIST>
+> = Boolean._$xnor<
+  A_SGN extends '+' ? true : false,
+  B_SGN extends '+' ? true : false
+> extends true
+  ? QUOTIENT
+  : Number._$negate<QUOTIENT>
```
Comment (poteat) [inline src/integer/divide.ts:24]: Exceedingly elegant solution!

**Comment by poteat** [comment]
This is so exciting! It seems this is basically ready for release :D

**Comment by MajorLift** [comment]
I think it's as ready as it's going to be haha. 
Next stop is `DecimalDigitList` and `Number.{Add,Subtract,Multiply}`, which should make it easier to attempt `Fractional`.

**Comment by MajorLift** [comment]
@poteat Fixed the issue with the stress tests! Should be ready to go now.

**Pull request merged**